cabal-testsuite/src/Test/Cabal/Monad.hs

```diff
@@ -232,7 +232,11 @@ runTestM mode m = withSystemTempDirectory "cabal-testsuite" $ \tmp_dir -> do
     args <- execParser (info testArgParser Data.Monoid.mempty)
     let dist_dir = testArgDistDir args
         (script_dir0, script_filename) = splitFileName (testArgScriptPath args)
-        script_base = dropExtensions script_filename
+
+        stripped = stripExtension ".test.hs" script_filename
+               <|> stripExtension ".multitest.hs" script_filename
```

Comment:
If you prefer, you can just add `<|> Just (dropExtension script_filename)` at the end of `stripped`, I think.